File: app/src/main/java/com/lifedawn/bestweather/weathers/simplefragment/currentconditions/SimpleCurrentConditionsFragment.java
package com.lifedawn.bestweather.weathers.simplefragment.currentconditions;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bumptech.glide.Glide;
import com.lifedawn.bestweather.R;
import com.lifedawn.bestweather.commons.enums.ValueUnits;
import com.lifedawn.bestweather.commons.enums.WeatherDataType;
import com.lifedawn.bestweather.main.MyApplication;
import com.lifedawn.bestweather.retrofit.responses.aqicn.AqiCnGeolocalizedFeedResponse;
import com.lifedawn.bestweather.weathers.WeatherFragment;
import com.lifedawn.bestweather.weathers.dataprocessing.response.AqicnResponseProcessor;
import com.lifedawn.bestweather.weathers.dataprocessing.util.LocationDistance;
import com.lifedawn.bestweather.weathers.dataprocessing.util.WeatherUtil;
import com.lifedawn.bestweather.weathers.models.AirQualityDto;
import com.lifedawn.bestweather.weathers.models.CurrentConditionsDto;
import com.lifedawn.bestweather.weathers.simplefragment.base.BaseSimpleCurrentConditionsFragment;
import com.lifedawn.bestweather.weathers.viewmodels.WeatherViewModel;

import org.jetbrains.annotations.NotNull;

import java.time.ZonedDateTime;

public class SimpleCurrentConditionsFragment extends BaseSimpleCurrentConditionsFragment {
	private static CurrentConditionsDto currentConditionsDto;
	private static AirQualityDto airQualityDto;

	public SimpleCurrentConditionsFragment setAirQualityDto(AirQualityDto airQualityDto) {
		this.airQualityDto = airQualityDto;
		return this;
	}

	public SimpleCurrentConditionsFragment setCurrentConditionsDto(CurrentConditionsDto currentConditionsDto) {
		this.currentConditionsDto = currentConditionsDto;
		return this;
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		return super.onCreateView(inflater, container, savedInstanceState);
	}

	@Override
	public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
	}


	@Override
	public void onInflateFinished(@NonNull View view, int resid, @Nullable ViewGroup parent) {
		super.onInflateFinished(view, resid, parent);
		setValuesToViews();

	}

	@Override
	public void setValuesToViews() {
		if (currentConditionsDto.isHasPrecipitationVolume()) {
			String precipitation = getString(R.string.precipitation_volume) + " : " + currentConditionsDto.getPrecipitationVolume();
			binding.precipitation.setText(precipitation);
		} else {
			binding.precipitation.setVisibility(View.GONE);
		}

		if (currentConditionsDto.getWindDirection() != null) {
			binding.windDirectionArrow.setRotation(currentConditionsDto.getWindDirectionDegree() + 180);
		}
		binding.windDirectionArrow.setVisibility(currentConditionsDto.getWindDirection() == null ? View.GONE : View.VISIBLE);
		binding.windDirection.setVisibility(currentConditionsDto.getWindDirection() == null ? View.GONE : View.VISIBLE);

		Glide.with(binding.weatherIcon).load(currentConditionsDto.getWeatherIcon()).into(binding.weatherIcon);
		binding.sky.setText(currentConditionsDto.getWeatherDescription());
		binding.wind.setText(currentConditionsDto.getWindStrength() != null ? currentConditionsDto.getWindStrength() :
				getString(R.string.noWindData));

		binding.windDirection.setText(currentConditionsDto.getWindDirection());

		binding.humidity.setText(String.format("%s %s", getString(R.string.humidity), currentConditionsDto.getHumidity()));

		final String tempUnitStr = MyApplication.VALUE_UNIT_OBJ.getTempUnitText();
		final String currentTempText = currentConditionsDto.getTemp().replace(tempUnitStr, "");

		binding.temperature.setText(currentTempText);
		binding.tempUnit.setText(tempUnitStr);

		binding.feelsLikeTemp.setText(currentConditionsDto.getFeelsLikeTemp().replace(tempUnitStr, ""));
		binding.feelsLikeTempUnit.setText(tempUnitStr);

		if (currentConditionsDto.getYesterdayTemp() != null) {
			binding.tempDescription.setText(WeatherUtil.makeTempCompareToYesterdayText(currentConditionsDto.getTemp(),
					currentConditionsDto.getYesterdayTemp(), tempUnit, requireContext().getApplicationContext()));
			binding.tempDescription.setVisibility(View.VISIBLE);
		} else {
			binding.tempDescription.setVisibility(View.GONE);
		}

		String airQuality = null;

		if (airQualityDto.isSuccessful()) {
			final double distance = LocationDistance.distance(latitude, longitude, airQualityDto.getLatitude(), airQualityDto.getLongitude(),
					LocationDistance.Unit.KM);

			if (distance > 100.0) {
				airQuality = getString(R.string.noData);
			} else {
				airQuality = AqicnResponseProcessor.getGradeDescription(airQualityDto.getAqi());
			}
		} else {
			airQuality = getString(R.string.noData);
		}

		binding.airQuality.setText(airQuality);

	}

}

File: app/src/main/java/com/lifedawn/bestweather/ui/weathers/simple/currentconditions/SimpleCurrentConditionsFragment.kt
package com.lifedawn.bestweather.ui.weathers.simple.currentconditions

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.lifedawn.bestweather.R
import com.lifedawn.bestweather.data.local.weather.models.AirQualityDto
import com.lifedawn.bestweather.data.local.weather.models.CurrentConditionsDto
import com.lifedawn.bestweather.data.local.weather.models.DailyForecastDto.Values.isHasPrecipitationVolume
import com.lifedawn.bestweather.data.remote.weather.aqicn.AqicnResponseProcessor.getGradeDescription
import com.lifedawn.bestweather.data.remote.weather.dataprocessing.util.LocationDistance
import com.lifedawn.bestweather.data.remote.weather.dataprocessing.util.LocationDistance.distance
import com.lifedawn.bestweather.data.remote.weather.dataprocessing.util.WeatherUtil.makeTempCompareToYesterdayText
import com.lifedawn.bestweather.ui.weathers.simple.base.BaseSimpleCurrentConditionsFragment

class SimpleCurrentConditionsFragment : BaseSimpleCurrentConditionsFragment() {
    fun setAirQualityDto(airQualityDto: AirQualityDto): SimpleCurrentConditionsFragment {
        this.airQualityDto = airQualityDto
        return this
    }

    fun setCurrentConditionsDto(currentConditionsDto: CurrentConditionsDto): SimpleCurrentConditionsFragment {
        this.currentConditionsDto = currentConditionsDto
        return this
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }

    override fun onInflateFinished(view: View, resid: Int, parent: ViewGroup?) {
        super.onInflateFinished(view, resid, parent)
        setValuesToViews()
    }

    override fun setValuesToViews() {
        if (currentConditionsDto.isHasPrecipitationVolume()) {
            val precipitation = getString(R.string.precipitation_volume) + " : " + currentConditionsDto!!.precipitationVolume
            binding!!.precipitation.text = precipitation
        } else {
            binding!!.precipitation.visibility = View.GONE
        }
        if (currentConditionsDto!!.windDirection != null) {
            binding!!.windDirectionArrow.rotation =
                (currentConditionsDto.windDirectionDegree + 180).toFloat()
        }
        binding!!.windDirectionArrow.visibility =
            if (currentConditionsDto.windDirection == null) View.GONE else View.VISIBLE
        binding!!.windDirection.visibility =
            if (currentConditionsDto.windDirection == null) View.GONE else View.VISIBLE
        Glide.with(binding!!.weatherIcon).load(currentConditionsDto.getWeatherIcon()).into(binding!!.weatherIcon)
        binding!!.sky.text = currentConditionsDto.weatherDescription
        binding!!.wind.text =
            if (currentConditionsDto.windStrength != null) currentConditionsDto.windStrength else getString(
                R.string.noWindData
            )
        binding!!.windDirection.text = currentConditionsDto.windDirection
        binding!!.humidity.text = String.format(
            "%s %s",
            getString(R.string.humidity),
            currentConditionsDto.humidity
        )
        val tempUnitStr = MyApplication.VALUE_UNIT_OBJ.tempUnitText
        val currentTempText = currentConditionsDto.temp.replace(tempUnitStr, "")
        binding!!.temperature.text = currentTempText
        binding!!.tempUnit.text = tempUnitStr
        binding!!.feelsLikeTemp.text = currentConditionsDto.feelsLikeTemp.replace(tempUnitStr, "")
        binding!!.feelsLikeTempUnit.text = tempUnitStr
        if (currentConditionsDto.getYesterdayTemp() != null) {
            binding!!.tempDescription.text = makeTempCompareToYesterdayText(
                currentConditionsDto.temp,
                currentConditionsDto.getYesterdayTemp(), tempUnit, requireContext().applicationContext
            )
            binding!!.tempDescription.visibility = View.VISIBLE
        } else {
            binding!!.tempDescription.visibility = View.GONE
        }
        var airQuality: String? = null
        airQuality = if (airQualityDto.isSuccessful()) {
            val distance = distance(
                latitude!!,
                longitude!!,
                airQualityDto!!.latitude,
                airQualityDto.longitude,
                LocationDistance.Unit.KM
            )
            if (distance > 100.0) {
                getString(R.string.noData)
            } else {
                getGradeDescription(airQualityDto.aqi)
            }
        } else {
            getString(R.string.noData)
        }
        binding!!.airQuality.text = airQuality
    }

    companion object {
        private val currentConditionsDto: CurrentConditionsDto? = null
        private val airQualityDto: AirQualityDto? = null
    }
}